logmerge.py

```diff
@@ -154,6 +155,8 @@ def render(line, prefix_arg=None, color=-1):
     :param str prefix_arg: Optional prefix to be stuck at the beginning of the rendered line
     :param int color: If 0-255, insert escape codes to display the line in that color
     """
+    if not sys.stdout.isatty():
```

Comment:
This is too expensive a call to make for every line of output. The answer cannot change during program execution, so we only need to determine this value once.

The best place to make this change is back in main() when the color[] array is initialized. Line 186 sets a default "color" to be 15 if colorization is disabled or the code runs out of colors. Refactor that hard-coded 15 into a "default_color" variable before the `for path in args.logfiles` loop starts, and make the default -1 if isatty() is false.